### ISSUE TITLE
Compilation on Mac OS X 10.10.3 / Apple clang 6.0 fails

### ISSUE THREAD

**Issue by felixbuenemann** [tatsuhiro-t/nghttp2#183]
Current master 98add63cdfabdbefc3c613d648762a9a01344a88 cannot be built on OS X Yosemite using clang from Xcode 6.2 (0.7.5 works):
```
uname -a
Darwin couchbook.lan 14.3.0 Darwin Kernel Version 14.3.0: Thu Feb 12 18:38:33 PST 2015; root:xnu-2782.20.34~3/RELEASE_X86_64 x86_64

clang++ --version
Apple LLVM version 6.0 (clang-600.0.57) (based on LLVM 3.5svn)
Target: x86_64-apple-darwin14.3.0
Thread model: posix

libtool: compile:  clang++ -DHAVE_CONFIG_H -I. -I.. -Wall -I../lib/includes -I../lib/includes -I../lib -I../src/includes -I../third-party -I/usr/local/Cellar/spdylay/1.3.2/include -I/usr/local/Cellar/libxml2/2.9.2/include/libxml2 -I/usr/local/Cellar/openssl/1.0.2/include -I/usr/local/Cellar/jansson/2.7/include -DHAVE_CONFIG_H -pthread -I/usr/local/opt/boost/include -g -O2 -MT libnghttp2_asio_la-asio_server_http2_handler.lo -MD -MP -MF .deps/libnghttp2_asio_la-asio_server_http2_handler.Tpo -c asio_server_http2_handler.cc  -fno-common -DPIC -o .libs/libnghttp2_asio_la-asio_server_http2_handler.o
In file included from asio_server_http2_impl.cc:25:
In file included from ./asio_server_http2_impl.h:30:
../src/includes/nghttp2/asio_http2_server.h:74:56: error: chosen constructor is explicit in copy-initialization
  void write_head(unsigned int status_code, header_map h = {}) const;
                                                       ^   ~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/map:1569:14: note: constructor declared here
    explicit multimap(const key_compare& __comp = key_compare())
             ^
../src/includes/nghttp2/asio_http2_server.h:74:56: note: passing argument to parameter 'h' here
  void write_head(unsigned int status_code, header_map h = {}) const;
                                                       ^
../src/includes/nghttp2/asio_http2_server.h:106:63: error: chosen constructor is explicit in copy-initialization
                       std::string raw_path_query, header_map h = {}) const;
                                                              ^   ~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/map:1569:14: note: constructor declared here
    explicit multimap(const key_compare& __comp = key_compare())
             ^
../src/includes/nghttp2/asio_http2_server.h:106:63: note: passing argument to parameter 'h' here
                       std::string raw_path_query, header_map h = {}) const;
                                                              ^
In file included from asio_server_http2.cc:27:
../src/includes/nghttp2/asio_http2_server.h:74:56: error: chosen constructor is explicit in copy-initialization
  void write_head(unsigned int status_code, header_map h = {}) const;
                                                       ^   ~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/map:1569:14: note: constructor declared here
    explicit multimap(const key_compare& __comp = key_compare())
             ^
../src/includes/nghttp2/asio_http2_server.h:74:56: note: passing argument to parameter 'h' here
  void write_head(unsigned int status_code, header_map h = {}) const;
                                                       ^
../src/includes/nghttp2/asio_http2_server.h:106:63: error: chosen constructor is explicit in copy-initialization
                       std::string raw_path_query, header_map h = {}) const;
                                                              ^   ~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/map:1569:14: note: constructor declared here
    explicit multimap(const key_compare& __comp = key_compare())
             ^
../src/includes/nghttp2/asio_http2_server.h:106:63: note: passing argument to parameter 'h' here
                       std::string raw_path_query, header_map h = {}) const;
                                                              ^
In file included from asio_server_http2_handler.cc:25:
In file included from ./asio_server_http2_handler.h:36:
../src/includes/nghttp2/asio_http2_server.h:74:56: error: chosen constructor is explicit in copy-initialization
  void write_head(unsigned int status_code, header_map h = {}) const;
                                                       ^   ~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/map:1569:14: note: constructor declared here
    explicit multimap(const key_compare& __comp = key_compare())
             ^
../src/includes/nghttp2/asio_http2_server.h:74:56: note: passing argument to parameter 'h' here
  void write_head(unsigned int status_code, header_map h = {}) const;
                                                       ^
../src/includes/nghttp2/asio_http2_server.h:106:63: error: chosen constructor is explicit in copy-initialization
                       std::string raw_path_query, header_map h = {}) const;
                                                              ^   ~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/map:1569:14: note: constructor declared here
    explicit multimap(const key_compare& __comp = key_compare())
             ^
../src/includes/nghttp2/asio_http2_server.h:106:63: note: passing argument to parameter 'h' here
                       std::string raw_path_query, header_map h = {}) const;
                                                              ^
In file included from asio_server.cc:37:
In file included from ./asio_server.h:48:
../src/includes/nghttp2/asio_http2_server.h:74:56: error: chosen constructor is explicit in copy-initialization
  void write_head(unsigned int status_code, header_map h = {}) const;
                                                       ^   ~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/map:1569:14: note: constructor declared here
    explicit multimap(const key_compare& __comp = key_compare())
             ^
../src/includes/nghttp2/asio_http2_server.h:74:56: note: passing argument to parameter 'h' here
  void write_head(unsigned int status_code, header_map h = {}) const;
                                                       ^
../src/includes/nghttp2/asio_http2_server.h:106:63: error: chosen constructor is explicit in copy-initialization


                       std::string raw_path_query, header_map h = {}) const;
                                                              ^   ~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/map:1569:14: note: constructor declared here
    explicit multimap(const key_compare& __comp = key_compare())
             ^
../src/includes/nghttp2/asio_http2_server.h:106:63: note: passing argument to parameter 'h' here
                       std::string raw_path_query, header_map h = {}) const;
                                                              ^
In file included from asio_server_http2_handler.cc:33:
./asio_server_response_impl.h:49:56: error: chosen constructor is explicit in copy-initialization
  void write_head(unsigned int status_code, header_map h = {});
                                                       ^   ~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/map:1569:14: note: constructor declared here
    explicit multimap(const key_compare& __comp = key_compare())
             ^
./asio_server_response_impl.h:49:56: note: passing argument to parameter 'h' here
  void write_head(unsigned int status_code, header_map h = {});
                                                       ^
./asio_server_response_impl.h:59:57: error: chosen constructor is explicit in copy-initialization
                 std::string raw_path_query, header_map h = {}) const;
                                                        ^   ~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/map:1569:14: note: constructor declared here
    explicit multimap(const key_compare& __comp = key_compare())
             ^
./asio_server_response_impl.h:59:57: note: passing argument to parameter 'h' here
                 std::string raw_path_query, header_map h = {}) const;
```

**Comment by tatsuhiro-t** [comment]
I committed possible fix via 8ea26fd.  Could you try that?

**Comment by felixbuenemann** [comment]
Thanks, it works fine now!
